pennylane/qchem/vibrational/bosonic_mapping.py

```diff
@@ -0,0 +1,157 @@
+
+from functools import singledispatch
+from typing import Union
+
+import numpy as np
+import pennylane as qml
+from pennylane.pauli import PauliSentence, PauliWord
+
+from .bosonic import BoseSentence, BoseWord
+
+
+def _get_pauli_op(i, j, qub_id):
+    r"""Returns expression to convert qubit-local term ::math::``\ket{x_j}\bra{x^{'}}``
+    to qubit operators as given in Eq. (6-9) <https://www.nature.com/articles/s41534-020-0278-0>"""
+
+    c1, c2 = 0.5, -0.5 if i == 1 else 0.5
+
+    if i != j:
+        return PauliSentence({PauliWord({qub_id: "X"}): c1, PauliWord({qub_id: "Y"}): c2 * 1j})
+
+    return PauliSentence({PauliWord({}): c1, PauliWord({qub_id: "Z"}): c2})
+
+
+def binary_mapping(
+    bose_operator: Union[BoseWord, BoseSentence],
+    nstates_boson: int = 2,
+    ps: bool = True,
+    wire_map: dict = None,
+    tol: float = None,
+):
+    r"""Convert a bosonic operator to a qubit operator using the standard-binary mapping
+    as described in `arXiv:1507.03271 <https://arxiv.org/pdf/1507.03271>`_.
+
+    Args:
+      bose_operator(BoseWord, BoseSentence): the bosonic operator
+      nstates_boson(int): Number of states in the boson.
+      ps (bool): whether to return the result as a PauliSentence instead of an operator. Defaults to False.
+      wire_map (dict): a dictionary defining how to map the orbitals of
+      the Bose operator to qubit wires. If None, the integers used to
+      order the orbitals will be used as wire labels. Defaults to None.
+      tol (float): tolerance for discarding the imaginary part of the coefficients
+
+    Returns:
+      a linear combination of qubit operators
+
+    **Example**
+
+    >>> w = qml.labs.vibrational.BoseWord({(0, 0): "+"})
+    >>> binary_mapping(w, nstates_boson=4)
+    0.6830127018922193 * X(0)
+    + -0.1830127018922193 * X(0) @ Z(1)
+    + -0.6830127018922193j * Y(0)
+    + 0.1830127018922193j * Y(0) @ Z(1)
+    + 0.3535533905932738 * X(0) @ X(1)
+    + -0.3535533905932738j * X(0) @ Y(1)
+    + 0.3535533905932738j * Y(0) @ X(1)
+    + (0.3535533905932738+0j) * Y(0) @ Y(1)
+    """
+
+    qubit_operator = _binary_mapping_dispatch(bose_operator, nstates_boson, tol=tol)
+
+    wires = list(bose_operator.wires) or [0]
+    identity_wire = wires[0]
+    if not ps:
+        qubit_operator = qubit_operator.operation(wire_order=[identity_wire])
+
+    if wire_map:
+        return qubit_operator.map_wires(wire_map)
+
+    return qubit_operator
+
+
+@singledispatch
+def _binary_mapping_dispatch(bose_operator, nstates_boson):
+    """Dispatches to appropriate function if bose_operator is a BoseWord or BoseSentence."""
+    raise ValueError(f"bose_operator must be a BoseWord or BoseSentence, got: {bose_operator}")
+
+
+@_binary_mapping_dispatch.register
+def _(bose_operator: BoseWord, nstates_boson, tol=None):
+
+    if nstates_boson < 2:
+        raise ValueError(
+            f"Number of bosonic states cannot be less than 2, provided {nstates_boson}."
```

Comment:
Why it cannot be less than 1?

Reply:
Binary mapping uses states (0 and 1) of the qubit to represent the state that the boson occupies, so that is the minimum number we need. For more than 2 states, we use multiple qubits to encode the states, for example 
For 3 levels: 
level 1 → |00⟩
level 2 → |01⟩
level 3 → |10⟩